testutil/isstrictnilchecker.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package testutil
+
+import (
+	"fmt"
+	"reflect"
+
+	"gopkg.in/check.v1"
+)
+
+// IsStrictNil checks that the variable is nil typed and nil valued.
+var IsStrictNil = &isStrictNilChecker{
+	&check.CheckerInfo{Name: "IsStrictNil", Params: []string{"value"}},
+}
+
+type isStrictNilChecker struct {
+	*check.CheckerInfo
+}
+
+func (*isStrictNilChecker) Check(params []interface{}, names []string) (result bool, errMsg string) {
+	if reflect.ValueOf(params[0]).IsValid() {
+		return false, fmt.Sprintf("expected <nil> type but got %T", params[0])
+	}
+
+	return true, ""
+}
+
+// StrictlyNotNil checks that the variable is not nil typed and nil valued.
+var StrictlyNotNil = &strictlyNotNilChecker{
```

Comment:
Nitpick, and only valid if we stay with this naming
```suggestion
var StrictNotNil = &strictNotNilChecker{
```